File: demo/src/main/java/com/devbrackets/android/recyclerextdemo/ui/viewholder/SimpleDragItemViewHolder.java
package com.devbrackets.android.recyclerextdemo.ui.viewholder;

import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.devbrackets.android.recyclerextdemo.R;

/**
 * A ViewHolder for the list_item_simple_text layout
 */
public class SimpleDragItemViewHolder extends RecyclerView.ViewHolder {
    private TextView textView;

    public static SimpleDragItemViewHolder newInstance(LayoutInflater inflater, ViewGroup parent) {
        View view = inflater.inflate(R.layout.list_item_simple_drag, parent, false);
        return new SimpleDragItemViewHolder(view);
    }

    public SimpleDragItemViewHolder(View itemView) {
        super(itemView);

        textView = itemView.findViewById(R.id.simple_drag_item_text_view);
    }

    public void setText(String text) {
        textView.setText(text);
    }

    public void setPosition(int position) {
        textView.setTag(position);
    }

    public void setOnClickListener(View.OnClickListener listener) {
        textView.setOnClickListener(listener);
    }

    public void setOnLongClickListener(View.OnLongClickListener listener) {
        textView.setOnLongClickListener(listener);
    }

    public void setOnTouchListener(View.OnTouchListener listener) {
        textView.setOnTouchListener(listener);
    }
}

File: demo/src/main/kotlin/com/devbrackets/android/recyclerextdemo/ui/viewholder/SimpleDragItemViewHolder.kt
package com.devbrackets.android.recyclerextdemo.ui.viewholder

import android.view.LayoutInflater
import android.view.View
import android.view.View.OnTouchListener
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.devbrackets.android.recyclerextdemo.R

/**
 * A ViewHolder for the list_item_simple_text layout
 */
class SimpleDragItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val textView: TextView
    fun setText(text: String?) {
        textView.text = text
    }

    fun setPosition(position: Int) {
        textView.tag = position
    }

    fun setOnClickListener(listener: View.OnClickListener?) {
        textView.setOnClickListener(listener)
    }

    fun setOnLongClickListener(listener: View.OnLongClickListener?) {
        textView.setOnLongClickListener(listener)
    }

    fun setOnTouchListener(listener: OnTouchListener?) {
        textView.setOnTouchListener(listener)
    }

    companion object {
        fun newInstance(inflater: LayoutInflater, parent: ViewGroup?): SimpleDragItemViewHolder {
            val view = inflater.inflate(R.layout.list_item_simple_drag, parent, false)
            return SimpleDragItemViewHolder(view)
        }
    }

    init {
        textView = itemView.findViewById(R.id.simple_drag_item_text_view)
    }
}